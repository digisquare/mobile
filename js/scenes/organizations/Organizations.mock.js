const Organizations = [{
  Organization: {
    id: '35',
    venue_id: '11',
    edition_id: '9',
    name: 'bdx.io',
    type: '0',
    avatar: 'https://pbs.twimg.com/profile_images/465909547752632320/ffD7qZ--_400x400.png',
    description: 'Une journée de conférences sur le développement informatique et les technologies de demain, qui aime bien les puissances de 2 avec près de 512 participants à guichets fermés. La prochaine édition aura lieu le 21 octobre 2016.',
    contacts: '{"twitter":"bdxio","facebook":"bdxio","website":"http:\\/\\/www.bdx.io\\/","societe":""}',
    event_count: '3',
    recent_event_count: '1',
    created: '2015-01-27 21:01:44',
    modified: '2016-11-27 04:00:01',
    Contacts: {
      twitter: 'bdxio',
      facebook: 'bdxio',
      website: 'http://www.bdx.io/',
      societe: '',
    },
  },
  Venue: {
    id: '11',
    edition_id: '9',
    name: 'Enseirb Matmeca',
    address: '1 Avenue Du Dr Albert Schweitzer',
    zipcode: '33400',
    city: 'Talence',
    country_code: 'FR',
    latitude: '44.802614',
    longitude: '-0.588054',
    event_count: '50',
    created: '2015-01-24 18:05:08',
    modified: '2016-05-26 14:03:21',
    oneliner: 'Enseirb Matmeca, 1 Avenue Du Dr Albert Schweitzer, 33400 Talence, FR',
  },
  Edition: {
    id: '9',
    name: 'Bordeaux',
    slug: 'bordeaux',
    event_count: '906',
    organization_count: '0',
    created: '2015-01-24 13:35:02',
    modified: '2015-01-24 13:35:02',
  },
}, {
  Organization: {
    id: '15',
    venue_id: '7',
    edition_id: '9',
    name: 'D&B Talks',
    type: '0',
    avatar: 'https://pbs.twimg.com/profile_images/3238741776/26f091b7287359733e08a454f10bbe87_400x400.png',
    description: "D&B Talks c'est le rendez-vous de tous les designers et pros du numérique. On y parle UX, UI, Web, Mobile, motion design...",
    contacts: '{"twitter":"dbtalks_org","facebook":"dbtalksbdx","website":"http:\\/\\/dbtalks.org\\/"}',
    event_count: '20',
    recent_event_count: '0',
    created: '2015-01-24 13:35:25',
    modified: '2016-11-07 04:00:01',
    Contacts: {
      twitter: 'dbtalks_org',
      facebook: 'dbtalksbdx',
      website: 'http://dbtalks.org/',
    },
  },
  Venue: {
    id: '7',
    edition_id: '9',
    name: "The Ramblin' Man",
    address: '52 Quai Richelieu',
    zipcode: '33000',
    city: 'Bordeaux',
    country_code: 'FR',
    latitude: '44.837321',
    longitude: '-0.566730',
    event_count: '28',
    created: '2015-01-24 18:05:06',
    modified: '2015-01-24 20:58:51',
    oneliner: "The Ramblin' Man, 52 Quai Richelieu, 33000 Bordeaux, FR",
  },
  Edition: {
    id: '9',
    name: 'Bordeaux',
    slug: 'bordeaux',
    event_count: '906',
    organization_count: '0',
    created: '2015-01-24 13:35:02',
    modified: '2015-01-24 13:35:02',
  },
}, {
  Organization: {
    id: '22',
    venue_id: null,
    edition_id: '9',
    name: 'BordeauxJS',
    type: '0',
    avatar: 'https://pbs.twimg.com/profile_images/2741451195/8ff14ab4849d749af06271795a15cf54_400x400.png',
    description: 'User group JavaScript à #bordeaux : réunions, discussions et live coding entre gens en asynchrone. #nodejs #coffeescript #javascript',
    contacts: '{"twitter":"BordeauxJS","website":"http:\\/\\/www.meetup.com\\/BordeauxJS\\/"}',
    event_count: '7',
    recent_event_count: '0',
    created: '2015-01-24 13:35:25',
    modified: '2016-07-22 04:00:01',
    Contacts: {
      twitter: 'BordeauxJS',
      website: 'http://www.meetup.com/BordeauxJS/',
    },
  },
  Venue: {
    id: null,
    edition_id: null,
    name: null,
    address: null,
    zipcode: null,
    city: null,
    country_code: null,
    latitude: null,
    longitude: null,
    event_count: null,
    created: null,
    modified: null,
    oneliner: null,
  },
  Edition: {
    id: '9',
    name: 'Bordeaux',
    slug: 'bordeaux',
    event_count: '906',
    organization_count: '0',
    created: '2015-01-24 13:35:02',
    modified: '2015-01-24 13:35:02',
  },
}];

export default Organizations;
