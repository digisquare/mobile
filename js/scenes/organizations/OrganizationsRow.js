import React, { PropTypes } from 'react';
import { Text, View, Image, StyleSheet, TouchableOpacity } from 'react-native';
import { Answers } from 'react-native-fabric';

import Router from '../../router';
import DigiColors from '../../common/DigiColors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    padding: 10,
  },
  avatarContainer: {
    width: 50,
  },
  avatar: {
    width: 40,
    height: 40,
    marginTop: 3,
  },
  textContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
  },
  title: {
    fontSize: 18,
    fontWeight: '900',
    color: DigiColors.secondaryFontColor,
  },
  venue: {
    color: DigiColors.secondaryFontColor,
  },
});

const selectOrganization = (navigator, organization) => {
  Answers.logContentView(
    `${organization.Organization.name}`,
    'organization',
    `organizations/${organization.Organization.id}`,
  );
  navigator.push(Router.getRoute('organization', { organization }));
};

export default function OrganizationsRow({ navigator, organization }) {
  return (
    <TouchableOpacity
      onPress={() => selectOrganization(navigator, organization)}
      style={styles.container}
    >
      <View style={styles.avatarContainer}>
        {
          organization.Organization.avatar ? (
            <Image
              style={styles.avatar}
              source={{ uri: organization.Organization.avatar }}
            />
          ) : null
        }
      </View>
      <View style={styles.textContainer}>
        <Text style={styles.title}>
          {organization.Organization.name}
        </Text>
        <Text style={styles.venue}>
          {
            organization.Venue && organization.Venue.name ? (
              `@ ${organization.Venue.name}`
            ) : null
          }
        </Text>
      </View>
    </TouchableOpacity>
  );
}

OrganizationsRow.propTypes = {
  navigator: PropTypes.object.isRequired,
  organization: PropTypes.object.isRequired,
};
