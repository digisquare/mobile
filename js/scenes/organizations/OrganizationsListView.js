import React, { Component, PropTypes } from 'react';
import { ListView, RefreshControl } from 'react-native';

import DigiError from '../../common/DigiError';

import OrganizationsRow from './OrganizationsRow';

import tumbeastNetwork from '../../common/img/tumbeast-network.png';
import tumbeastSitting from '../../common/img/tumbeast-sitting.png';

export default class OrganizationsListView extends Component {
  constructor(props) {
    super(props);
    this.renderOrganizationsRow = this.renderOrganizationsRow.bind(this);
  }

  renderOrganizationsRow(organization) {
    return (
      <OrganizationsRow
        navigator={this.props.navigator}
        organization={organization}
      />
    );
  }

  render() {
    const { dataSource, error, refreshing, onRefresh } = this.props;
    const refreshControl = (
      <RefreshControl
        refreshing={refreshing}
        onRefresh={onRefresh}
      />
    );

    if (error) {
      return (
        <DigiError
          refreshControl={refreshControl}
          image={tumbeastNetwork}
          text="Oups, impossible de récupérer les organisateurs"
        />
      );
    }

    if (dataSource.getRowCount() === 0 && !refreshing) {
      return (
        <DigiError
          refreshControl={refreshControl}
          image={tumbeastSitting}
          text="Snif, aucun organisateur dans cette ville"
        />
      );
    }

    return (
      <ListView
        dataSource={dataSource}
        refreshControl={refreshControl}
        renderRow={this.renderOrganizationsRow}
        enableEmptySections
      />
    );
  }
}

OrganizationsListView.propTypes = {
  navigator: PropTypes.object.isRequired,
  dataSource: PropTypes.object.isRequired,
  refreshing: PropTypes.bool.isRequired,
  onRefresh: PropTypes.func.isRequired,
  error: PropTypes.bool.isRequired,
};
