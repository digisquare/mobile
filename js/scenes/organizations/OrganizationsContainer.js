import React, { Component, PropTypes } from 'react';
import { View, ListView, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { Answers } from 'react-native-fabric';

import { fetchOrganizations } from '../../actions/organizations';

import OrganizationsListView from './OrganizationsListView';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

class OrganizationsContainer extends Component {
  static route = {
    navigationBar: {
      title({ title }) {
        return title || 'Bordeaux';
      },
    },
  }

  constructor(props) {
    super(props);
    const { editions: { selectedEdition }, organizations } = props;
    const items = organizations[selectedEdition]
      && organizations[selectedEdition].items
      && organizations[selectedEdition].items.length > 0;
    this.state = {
      dataSource: new ListView.DataSource({
        rowHasChanged: (row1, row2) => row1 !== row2,
      }),
      refreshing: !items,
      error: false,
    };
    this.onRefresh = this.onRefresh.bind(this);
  }

  componentWillMount() {
    const { onFetchOrganizations, editions, editions: { selectedEdition } } = this.props;
    const editionName = editions.items.find(edition => edition.id === selectedEdition).name;
    onFetchOrganizations(selectedEdition);
    Answers.logContentView(`Organisateurs Ã  ${editionName}`, 'organizations', 'organizations');
  }

  componentWillReceiveProps(nextProps) {
    const { onFetchOrganizations, organizations, editions } = nextProps;
    const { selectedEdition } = editions;
    const { dataSource, refreshing } = this.state;
    if (selectedEdition !== this.props.editions.selectedEdition) {
      const edition = editions.items.find(e => e.id === selectedEdition);
      this.props.navigator.updateCurrentRouteParams({
        title: edition.name,
      });
      onFetchOrganizations(selectedEdition);
      this.setState({
        refreshing: true,
      });
    }
    if (!organizations[selectedEdition]) {
      return this.setState({
        dataSource: dataSource.cloneWithRowsAndSections({}),
      });
    }
    const { error, isFetching, items } = organizations[selectedEdition];
    if (error && !isFetching && items.length === 0) {
      return this.setState({
        refreshing: false,
        error: true,
      });
    }
    if (refreshing && !isFetching) {
      this.setState({
        refreshing: false,
      });
    }
    if (items) {
      return this.setState({
        dataSource: dataSource.cloneWithRows(items),
        error: false,
      });
    }
    return null;
  }

  onRefresh() {
    const { onFetchOrganizations, editions: { selectedEdition } } = this.props;
    onFetchOrganizations(selectedEdition);
    return this.setState({
      refreshing: true,
    });
  }

  render() {
    const { navigator } = this.props;
    const { dataSource, refreshing, error } = this.state;
    return (
      <View style={styles.container}>
        <OrganizationsListView
          navigator={navigator}
          dataSource={dataSource}
          refreshing={refreshing}
          onRefresh={this.onRefresh}
          error={error}
        />
      </View>
    );
  }
}

const mapStateToProps = state => ({
  editions: state.editions,
  organizations: state.organizations,
});

const mapDispatchToProps = dispatch => ({
  onFetchOrganizations: (selectedEdition) => {
    dispatch(fetchOrganizations(selectedEdition));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(OrganizationsContainer);

OrganizationsContainer.propTypes = {
  navigator: PropTypes.object.isRequired,
  onFetchOrganizations: PropTypes.func.isRequired,
  editions: PropTypes.object.isRequired,
  organizations: PropTypes.object.isRequired,
};
