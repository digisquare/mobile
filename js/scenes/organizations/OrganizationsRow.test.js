import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import OrganizationsRow from './OrganizationsRow';

import organizations from './Organizations.mock';

it('renders correctly', () => {
  const tree = renderer.create(
    <OrganizationsRow navigator={{ push: () => null }} organization={organizations[0]} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
