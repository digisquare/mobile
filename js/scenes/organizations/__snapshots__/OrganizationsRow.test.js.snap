exports[`test renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "padding": 10,
    }
  }>
  <View
    style={
      Object {
        "width": 50,
      }
    }>
    <Image
      source={
        Object {
          "uri": "https://pbs.twimg.com/profile_images/465909547752632320/ffD7qZ--_400x400.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginTop": 3,
          "width": 40,
        }
      } />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "fontSize": 18,
          "fontWeight": "900",
        }
      }>
      bdx.io
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
        }
      }>
      @ Enseirb Matmeca
    </Text>
  </View>
</TouchableOpacity>
`;
